ibm/resource_ibm_cloudant_database.go

```diff
@@ -0,0 +1,189 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	rc "github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+)
+
+func resourceIbmCloudantDatabase() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIbmCloudantDatabaseCreate,
+		ReadContext:   resourceIbmCloudantDatabaseRead,
+		DeleteContext: resourceIbmCloudantDatabaseDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"instance_crn": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Cloudant Instance CRN.",
+			},
+			"db": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Path parameter to specify the database name.",
+			},
+			"partitioned": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Query parameter to specify whether to enable database partitions when creating a database.",
+			},
+			"q": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "The number of shards in the database. Each shard is a partition of the hash value range. Default is 8, unless overridden in the `cluster config`.",
```

Comment:
We should drop the default here like we're doing in the docs, it is not consistent. We should probably also add the "you are encouraged to talk to support about appropriate values before changing this" message.